iron-list.html

```diff
@@ -777,6 +831,25 @@
     },
 
     /**
+     * This function will return the amount the physical
+     * top should be incremented based on it's current iterative
+     * state in a given row. This function will only return a 
+     * non-zero value when the index modulous `_physicalItemsPerRow` 
+     * is one less than the possible `_physicalItemsPerRow`
+     */
+    _getPhysicalTopIncrement: function(kth) {
+      if (!Array.isArray(this.items)) return 0;
+      if (kth % this._physicalItemsPerRow !== (this._physicalItemsPerRow - 1)) return 0;
+      var rowSize = 0;
+      for (length = kth + this._physicalItemsPerRow,
+           maxLength = this.items.length;
+           kth < maxLength && kth < length; kth++) {
+        rowSize = this._physicalSizes[kth] > rowSize ? this._physicalSizes[kth] : rowSize;
+      }

```

Comment:
As it stands now the way that this estimates top position actually hasn't changed. Instead of estimating based on each individual item height the `_physicalAverage` is updated by row height.

While putting this together I was trying to think of a case where people had a true grid that had items of differing heights. I felt like it would like out of place. So I would be fine with both restricting usage to a fixed height API as well as centering each item in the row. 

It wouldn't be too difficult to center as we would just need to calculate a height offset per item in `_positionItems` based on the current row height. Not too bad either way.